wgengine/magicsock/magicsock.go

```diff
@@ -1581,6 +1581,9 @@ func (c *Conn) noteRecvActivityFromEndpoint(e conn.Endpoint) {
 
 // receiveIPv6 receives a UDP IPv6 packet. It is called by wireguard-go.
 func (c *Conn) receiveIPv6(b []byte) (int, conn.Endpoint, error) {
+	health.SetReceiveIPv6Started(true)
```

Comment:
I expected you to call this in connBind.Open, at:

```go
        if c.pconn6 != nil {
                fns = append(fns, c.receiveIPv6)
        }
```

Reply:
Then there's a (bigger) window during which you can get a false positive. Plus for anyone reading the code later, it makes it totally obvious that `started && !running` occurs when this function exits.

Reply:
Ah, okay, now I see.